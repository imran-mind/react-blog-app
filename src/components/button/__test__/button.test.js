import React from 'react';
import ReactDOM from 'react-dom';
import Button from './../button';
import {render} from '@testing-library/react'
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

// afterEach(cleanup)
it('renders without crashing ',()=>{
    const div = document.createElement('div');
    ReactDOM.render(<Button></Button>,div);
})

it('renders button correctly ',()=>{
    const {getByTestId} = render(<Button label="click me please"/>);
    expect(getByTestId('button')).toHaveTextContent('click me please')
})

it('renders save button correctly ',()=>{
    const {getByTestId} = render(<Button label="save"/>);
    expect(getByTestId('button')).toHaveTextContent('save')
})

it('matches snapshot save button',()=>{
    const tree = renderer.create(<Button label="save"/>).toJSON();
    expect(tree).toMatchSnapshot()
})
it('matches snapshot edit button',()=>{
    const tree = renderer.create(<Button label="edit"/>).toJSON();
    expect(tree).toMatchSnapshot()
})