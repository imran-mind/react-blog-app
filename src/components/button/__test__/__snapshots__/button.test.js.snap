// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot edit button 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  Hi 
  edit
</div>
`;

exports[`matches snapshot save button 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  Hi 
  save
</div>
`;
